src/coreclr/src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -81,7 +84,19 @@ internal object SyncRoot
             return codeBase;
         }
 
-        public override string? CodeBase => GetCodeBase(false);
+        public override string? CodeBase
+        {
+            get
+            {
+                var runtimeAssembly = this;
+                if (IsInBundle(new QCallAssembly(ref runtimeAssembly)))
```

Comment:
Do we really need a new QCall for this?
We could either throw from native (I know, that's tedious) or we could return null from the `GetCodeBase` QCall. As far as I can tell the native implementation of `GetCodeBase` will never return null.

I just don't want to introduce a new QCall if we don't need one (And also so far the managed parts of runtime were not aware of bundles).

Reply:
I agree that it is cleaner to do the check within `GetCodeBase()` in this case. 
We may eventually require the IsBundle() QCall API, but can sidestep it for now.